api/v1/resourcesetinputprovider_types.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2025 Stefan Prodan.
+// SPDX-License-Identifier: AGPL-3.0
+
+package v1
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/json"
+
+	"github.com/fluxcd/pkg/apis/meta"
+)
+
+const (
+	ResourceSetInputProviderKind    = "ResourceSetInputProvider"
+	GitHubPullRequestInputProvider  = "GitHubPullRequest"
+	GitLabMergeRequestInputProvider = "GitLabMergeRequest"
+)
+
+// ResourceSetInputProviderSpec defines the desired state of ResourceSetInputProvider
+type ResourceSetInputProviderSpec struct {
+	// Type specifies the type of the input provider.
+	// +kubebuilder:validation:Enum=GitHubPullRequest;GitLabMergeRequest
+	// +required
+	Type string `json:"type"`
+
+	// URL specifies the HTTP/S address of the input provider API.
+	// When connecting to a Git provider, the URL should point to the repository address.
+	// +kubebuilder:validation:Pattern="^(http|https)://.*$"
+	// +required
+	URL string `json:"url"`
+
+	// SecretRef specifies the Kubernetes Secret containing the basic-auth credentials
+	// to access the input provider. The secret must contain the keys
+	// 'username' and 'password'.
+	// When connecting to a Git provider, the password should be a personal access token
+	// that grants read-only access to the repository.
+	// +optional
+	SecretRef *meta.LocalObjectReference `json:"secretRef,omitempty"`
+
+	// CertSecretRef specifies the Kubernetes Secret containing either or both of
+	//
+	// - a PEM-encoded CA certificate (`ca.crt`)
+	// - a PEM-encoded client certificate (`tls.crt`) and private key (`tls.key`)
+	//
+	// When connecting to a Git provider that uses self-signed certificates, the CA certificate
+	// must be set in the Secret under the 'ca.crt' key to establish the trust relationship.
+	// +optional
+	CertSecretRef *meta.LocalObjectReference `json:"certSecretRef,omitempty"`
+
+	// DefaultValues contains the default values for the inputs.
+	// These values are used to populate the inputs when the provider
+	// response does not contain them.
+	// +optional
+	DefaultValues ResourceSetInput `json:"defaultValues,omitempty"`
```

Comment:
What kind of inputs are you expecting providers not to have? Do you have anything specific in mind?

Reply:
Anything app specific e.g. chart name, env, perms etc. See the docs for some example.